### PR TITLE
Ionization: Clean up Params

### PR DESCRIPTION
**Pull request by ax3l**
- move `atomicNumbers` to `ionizationConfig.param`
- rename `effectiveAtomicNumbers` to `effectiveNuclearCharge`
- add namespaces
- add Deuterium (aka H-2 or D), because it has a favorable charge-to-mass ratio for studying the dynamics of fully ionized hydrogen + non-hydrogen materials
- [ ] add gold (for @Heikman's example) -> follow up PR

### CODE REVIEW

File: src/picongpu/include/simulation_defines/param/ionizerConfig.param [hunk 1]
```diff
@@ -17,51 +17,158 @@
...
+ * - BSIStarkShifted : BSI for hydrogen-like atoms and ions considering the
+ *                     Stark upshift of ionization potentials
+ *
+ * Usage: Add flags to the list of particle flags that has the following structure
+ *
+ *        ionizer< particles::ionization::IonizationModel< Species2BCreated > >,
+ *        atomicNumbers< ionization::atomicNumbers::Element >,
+ *        effectiveNuclearCharge< ionization::effectiveNuclearCharge::ELEMENT_t >,
+ *        ionizationEnergies< ionization::energies::AU::ELEMENT_t >
+ */
```
Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/ionizerConfig.param]: new syntax above for discussion, we might want to make all with the same `Element_t` spelling?

Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/ionizerConfig.param]: offline discussion: @psychocoderHPC and @n01r agree :)

**Comment by ax3l** [comment]
If you like go ahead & merge. I can add gold in a follow up tomorrow.

File: src/picongpu/include/simulation_defines/param/ionizerConfig.param [hunk 2]
```diff
@@ -17,51 +17,158 @@
...
+    {
+        static constexpr float_X numberOfProtons  = 7.0;
+        static constexpr float_X numberOfNeutrons = 7.0;
+    };
+
+    /** O-16 99.76% NA */
+    struct Oxygen_t
+    {
+        static constexpr float_X numberOfProtons  = 8.0;
+        static constexpr float_X numberOfNeutrons = 16.0;
```
Comment (n01r) [inline src/picongpu/include/simulation_defines/param/ionizerConfig.param]: 8 Neutrons

Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/ionizerConfig.param]: thanks for reviewing!

Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/ionizerConfig.param]: offline discussion: ok, will fix

**Pull request merged**